### DISCUSSION TITLE
How can I know that all the tasks in the queue have completed?

### DISCUSSION THREAD

**Discussion by natilusky** [taskforcesh/bullmq#1459] [Q&A] [answered]
Hi,
I use bullmq to distribute WhatsApp messages to clients,
I would like to receive an event every time all the tasks in the queue have completed,

I tried to use
`    worker.on('drained', () => { console.error('drained'); });
`
But I receive the event twice when I use delay, the first time when it is waiting and the second time when everything has been completed, I want to receive once that everything has been completed.

I would appreciate your advice

**Comment by johnnyleejy** [comment, marked as answer]
You can check if the queue's completed job count === number of published messages

`if ((await queue.getCompletedCount()) === publishedCount){
      // do something
}`

**Reply by natilusky** [reply]
Thank you, it works = )

**Reply by johnnyleejy** [reply]
welcome :) You may mark this answered.
